meta-phosphor/conf/machine/include/obmc-bsp-common.inc

```diff
@@ -13,7 +13,7 @@ UBOOT_SUFFIX ?= "bin"
 
 MACHINEOVERRIDES =. "openbmc:"
 
-IMAGE_FSTYPES += "overlay"
+MACHINE_FEATURES += "image-overlay"
```

Comment:
On Jun 20, 2016, at 10:14 PM, Patrick Williams <notifications@github.com> wrote:
> 
> In meta-phosphor/conf/machine/include/obmc-bsp-common.inc:
> 
> > @@ -13,7 +13,7 @@ UBOOT_SUFFIX ?= "bin"
> >  
> >  MACHINEOVERRIDES =. "openbmc:"
> >  
> > -IMAGE_FSTYPES += "overlay"
> > +MACHINE_FEATURES += "image-overlay"
> 
> I understand the concern.
> 
> I think I originally created the "openpower-pflash" machine feature. I could make an argument that the "machine" runs the op-build firmware which requires pflash for management. But I could also be persuaded to use something else entirely.

No; I saw that logic too, which is why I didn’t say anything last time.

> Agree that overlay isn't really a machine feature. I think I was just following the first-found pattern.
> 
> Should I create arbitrary variable names? 

That is what I have been doing in other areas.  But we probably need to establish some conventions eventually.

> I don't know what the best approach is here. 'overlay' and 'openpower-pflash' do not seem like distro features. 'overlay' could be considered an image feature,

For clarity its not an image feature either.  Again, features of any kind determine what packages go in the rootfs.

> but it also needs some component of the machine with it at this point in time because qemu doesn't support the overlay partitioning. Looking for input.

When you say qemu do you mean the qemuarm machine or qemu in general?  There shouldn’t be a problem with qemuarm because it doesn’t pick up openpower.inc.  The problem is the core-image recipes on real hw right?

What we are doing here with the overlay is exactly what IMAGE_FSTYPES is there for.  I think the most correct thing to do would be to make the image-overlay class itself do the right thing when building a core image, whether that be degrading somehow, pointing at the right sub-images, or just not doing anything,  etc.

> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>